mailpoet/lib/Config/Migrator.php

```diff
@@ -932,4 +935,29 @@ private function migrateEmailActionsFilters(): bool {
     }
     return true;
   }
+
+  private function updateDefaultInactiveSubscriberTimeRange(): bool {
+    // only run if the current database version is older than 3.78.0
+    if (version_compare((string)$this->settings->get('db_version', '3.78.0'), '3.78.0', '>=')) {
+      return false;
+    }
+
+    $currentValue = (int)$this->settings->get('deactivate_subscriber_after_inactive_days');
+    if ($currentValue === 180) {
+      $this->settings->set('deactivate_subscriber_after_inactive_days', '365');
+
+      /**
+       * Ensure that inactive subscribers are recalculated as soon after the setting change as possible.
+       *
+       * This behavior is normally triggered when the setting is changed through the JSON API, i.e. when a user has
+       * changed the setting themselves through the UI. The `onInactiveSubscribersIntervalChange` method doesn't
+       * depend on any state, and calling it directly means avoiding code duplication.
+       */
+      $diContainer = ContainerWrapper::getInstance();
+      $apiSettings = $diContainer->get(Settings::class);
+      $apiSettings->onInactiveSubscribersIntervalChange();
```

Comment:
This doesn't feel like a great way of handling this. We're not using the DI container anywhere else in the migrator, and I'm not sure if there's a reason for that. Maybe more importantly than that, it seems like `onInactiveSubscribersIntervalChange` should live somewhere other than the `MailPoet\API\JSON\v1\Settings` class, but I don't think it makes sense to change that just for this migration.

There's also the possibility that the behavior of `\MailPoet\API\JSON\v1\Settings::onInactiveSubscribersIntervalChange` might change between when this is released and when a user actually runs the migration, so perhaps it would be better to duplicate whatever code necessary to ensure the migration's behavior stays consistent no matter what.

Reply:
In my opinion, you can refactor Migrator as a service. After that, when everything works well, you can inject the Settings class. But maybe it could be better to create a new service with this logic(only simple extract change events) to a new service. I guess we shouldn't inject the whole endpoint class.
I like your idea of avoiding code repetition. 👍🏻